lib/flowViewport/API/samples/fvpDataProducerSceneIndexExample.h

```diff
@@ -0,0 +1,187 @@
+//
+// Copyright 2023 Autodesk
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+#ifndef FLOW_VIEWPORT_EXAMPLES_DATA_PRODUCER_SCENE_INDEX_EXAMPLE_H
+#define FLOW_VIEWPORT_EXAMPLES_DATA_PRODUCER_SCENE_INDEX_EXAMPLE_H
+
+//Local headers
+#include "flowViewport/api.h"
+#include "flowViewport/API/fvpDataProducerSceneIndexInterface.h"
+
+//USD/Hydra headers
+#include <pxr/base/tf/refPtr.h>
+#include <pxr/base/gf/matrix4d.h>
+#include <pxr/base/vt/array.h>
+#include <pxr/imaging/hd/retainedSceneIndex.h>
+#include <pxr/imaging/hd/dataSource.h>
+#include <pxr/imaging/hd/retainedDataSource.h>
+
+namespace FVP_NS_DEF {
+
+/** This class is an example on how to inject Hydra prims into a Hydra viewport. 
+*   An data producer scene index is a scene index that adds primitives to the current rendering.
```

Comment:
Is it possible to add more details on how to use the DataProducerSceneIndexExample class in steps?
e.g., 
1.call setHydraInterface() to set the hydra scene index interface
2.call setContainerNode() to set a dcc node ...
3. call addDataProducerSceneIndex() to ...
...